src/templates/podcast.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
```

Comment:
I don't think we are running prop type checks anywhere right now, this was left over from the initial project fork. That being said, with typescript we might not need them anyway since we can also do that validation in TS. What do you think?

Reply:
Podcast.propTypes = {
    pageContext: PropTypes.shape({
        data: PropTypes.shape({
            description: PropTypes.string,
            rawDescription: PropTypes.string,
            publishDate: PropTypes.string,
            thumbnailUrl: PropTypes.string,
            title: PropTypes.string,
            url: PropTypes.string,
        }),
        slug: PropTypes.string,
    }),
}; 
this is being run for sure at the time of page creation.In this example the data I get back from podcasts which is of type Podcast may not have rawDescription , but for page creation, I definitely need the property. Sure I can create one more interface PodcastPage with mandatory rawDescription field, but that will mean more interfaces. I'm leaving it for now, but going forward , when I create new components in TS, I will start by not having prop type check and use interface.

Reply:
Did I answer your question?

Reply:
Ah okay yes thanks